modules/javafx.base/src/main/java/javafx/beans/property/ObjectPropertyBase.java

```diff
@@ -75,22 +88,22 @@ public ObjectPropertyBase(T initialValue) {
 
     @Override
     public void addListener(InvalidationListener listener) {
-        helper = ExpressionHelper.addListener(helper, this, listener);
+        LISTENER_MANAGER.addListener((ObjectPropertyBase<Object>) this, listener);
```

Comment:
I think the unchecked casts here can be removed if `ListenerManagerBase` is declared as `ListenerManagerBase<T, I extends ObservableValue<? extends T>>`, and `OldValueCachingListenerManager` accordingly. Then the `LISTENER_MANAGER` instance can be parameterized as `OldValueCachingListenerManager<Object, ObjectPropertyBase<?>>`.

Reply:
Thanks, I gave up on that one a bit, I was looking for a better solution, but never got the generics quite the way I wanted them there, but your change works. I was primarily aiming to keep the casts as much out of the inner loops as possible.

 I still need to do one cast at this line, but it is a huge improvement:

    @Override
    public void addListener(ChangeListener<? super T> listener) {
        LISTENER_MANAGER.addListener(this, (ChangeListener<Object>) listener);
    }